cdxev/auxiliary/output.py

```diff
@@ -99,13 +100,38 @@ def update_timestamp(sbom: dict) -> None:
 def update_tools(sbom: dict) -> None:
     """Adds this tool to the list of tools in metadata."""
     metadata: dict = sbom.setdefault("metadata", {})
-    tools: list = metadata.setdefault("tools", [])
-
-    this_tool = {
-        "name": pkg.NAME,
-        "vendor": pkg.VENDOR,
-        "version": pkg.VERSION,
-    }
+    spec_version = SpecVersion.parse(sbom.get("specVersion", ""))
+    tools: t.Optional[t.Union[dict, list]] = metadata.get("tools", None)
+
+    # Starting in CycloneDX 1.5 tools should be a dict.
+    if tools is None:
+        if spec_version is not None and spec_version >= CycloneDXVersion.V1_5:
```

Comment:
you are comparing ge, but have not defined it in the class

Reply:
That's the beaty of [dataclass](https://docs.python.org/3.11/library/dataclasses.html#dataclasses.dataclass) and [total_ordering()](https://docs.python.org/3.11/library/functools.html#functools.total_ordering). I don't have to implement all of the methods myself.